library/common/engine.h

```diff
@@ -31,12 +31,18 @@ class Engine : public Logger::Loggable<Logger::Id::main> {
    */
   ~Engine();
 
+  struct EngineRunOptions {
```

Comment:
👍

Reply:
I like this conceptually, but I do think perhaps config should be separate from it. There's probably more that can go in here down the road.